Remove legacy attribute: display="bitmask"

hamishwillee opened the pull request:
This removes the `display="bitmask"` attribute on fields that reference a bitmask enum from the ArduPilot-managed dialects. I believe that to be the entire included set:

- [x] ardupilotmega.xml
- [x] uAvionix.xml
- [x] icarous.xml - None
- [x] loweheiser.xml - None
- [x] cubepilot.xml - None
- [x] csAirLink.xml - None

All the enums that use this are marked up as bitmasks using `bitmask="true"`, so the information on the field is inferable. We're heading towards removing the bitmask indication on the field itself.

History here is that `display="bitmask"` was added on request of QGC to allow auto-generation of checkbox UI. We later decided that it was better to have on the enum, because that is the thing we know is a bitmask. 
However neither attribute is actually used in QGC or Mission Planner or MAVProxy. It might be used elsewhere, but unlikely.

Upstream removes them in https://github.com/mavlink/mavlink/pull/2215

This falls out of @IamPete1 consistency checker, added in https://github.com/mavlink/mavlink/pull/2206

tridge commented:
I thought display=bitmask wasn't originally for enums, it was for values to display as hex instead of decimal

tridge commented:
I see 10 fields that are marked as bitmask that are not marked with enum

amilcarlucas commented:
Please do not do this! This is very usefull for GCS software.

rmackay9 commented:
We discussed this on the weekly AP dev call and the consensus was that we couldn't merge this as-is because it was going to break lots of tools.

amilcarlucas commented:
Nope, let's close this.

IamPete1 commented:
> Please do not do this! This is very usefull for GCS software. Some enums are bitmasks, some are not!

This information is still available directly on the enum itself with the existing `bitmask=true` tag. The goal here is to remove this redundant mark-up and make it easier for the GCS to work out what is a bitmask and what is not.

IamPete1 commented:
> I thought display=bitmask wasn't originally for enums, it was for values to display as hex instead of decimal

I was looking into this, it turns out MAVLink also supports `print_format`, this seems like it would be a better fit for printing as hex use case. For example the sys status bitmasks uses `print_format="0x%04x"` (as well as `display="bitmask"` and `bitmask=true` on the enum)

hamishwillee commented:
The argument against seem to be invalid, though I might be missing something. Apologies - clearly my explanation/reasoning wasn't sufficient.

To be clear, no information is lost by this. We added `bitmask="true"` to the enums that these fields reference - the theory being that bitmask is a property of an enum and since enums are used in multiple messages, this is a more efficient than marking up the field.
This now removes the display information which is redundant.

@amilcarlucas @rmackay9 What tools will be broken by this? It was added originally for QGC to be able to auto render bitmasks as checkboxes. It has not been implemented by QGC (I checked). It is possible that it was supported by other tools - I somehow doubt it but would have to reconsider if there is any evidence - MP does not appear to.

> I see 10 fields that are marked as bitmask that are not marked with enum

@tridge I removed all cases where `display="bitmask"`. Checking again, all those cases have a corresponding enum marked up with `bitmask="true"`.

There is one case `MAG_CAL_PROGRESS` where no enum was listed in `cal_mask` - so information is lost, but it isn't clear whether or not this should have been a bitmak.

What are the other cases where fields are marked as a bitmask that I have not removed?

> I thought display=bitmask wasn't originally for enums, it was for values to display as hex instead of decimal

What @IamPete1 said above https://github.com/ArduPilot/mavlink/pull/383#issuecomment-2665500296 - I believe that is the `print_format` case - though this is the first time I've seen the explanation :-0

If we don't do this we should remove `bitmask="true"` - because we don't need to have the information twice. `bitmask="true"` is better, but whatever makes people happy.

amilcarlucas commented:
Thanks, now it is clear. If you prove that no information gets lost and that all bitmasks are marked as such, then I do not oppose to this.

hamishwillee commented:
>  If you prove that no information gets lost and that all bitmasks are marked as such, then I do not oppose to this.

We have (or at least I have tested that all the cases where this has been removed have corresponding enum updates).

IamPete1 reviewed (approved):
LGTM
